app/models/project.rb

```diff
@@ -516,6 +516,9 @@ def check_status(removed = false)
     response = Typhoeus.head(url)
     if platform.downcase == 'packagist' && response.response_code == 302
       update_attribute(:status, 'Removed')
+    elsif platform.downcase == 'pypi' && response.response_code == 404
+      # TODO: remove this stanza once this bug is fixed: https://github.com/pypa/warehouse/issues/3709#issuecomment-754973958
+      update_attribute(:status, 'Deprecated')
```

Comment:
I didn't add a test for this because it's temporary, but I did confirm that it works in the console.